pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -332,3 +335,29 @@ type DNSSDConfig struct {
 	// +optional
 	Port *int `json:"port"`
 }
+
+// EC2SDConfig allow retrieving scrape targets from AWS EC2 instances.
+// The private IP address is used by default, but may be changed to the public IP address with relabeling.
+// The IAM credentials used must have the ec2:DescribeInstances permission to discover scrape targets
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ec2_sd_config
+// +k8s:openapi-gen=true
+type EC2SDConfig struct {
+	// The AWS region. If blank, the region from the instance metadata is used
+	// +optional
```

Comment:
Since we are not implementing other fields like Profile, I am wondering if we should make this required till we implement those

Reply:
I'm not again since changing required fields to optional is not a breaking change :)

Reply:
I am keeping it as optional itself. I think I would need to make access key also required in that case. 
Open for suggestions how we want to do it